lib/Elastica/Multi/Search.php

```diff
@@ -27,18 +32,20 @@ class Search
     protected $_options = array();
 
     /**
-     * @var \Elastica\Client
+     * @var array|\Elastica\Search[]
      */
-    protected $_client;
+    protected $_searches = array();
 
     /**
      * Constructs search object.
      *
      * @param \Elastica\Client $client Client object
+     * @param MultiBuilderInterface $builder
      */
-    public function __construct(Client $client)
+    public function __construct(Client $client, MultiBuilderInterface $builder = null)
     {
-        $this->setClient($client);
+        $this->_builder = $builder ?: new MultiBuilder();
```

Comment:
I would suggest the following approach first:
* Client has no knowledge of the builders
* Each builder has its own default type

This is in a way identical to the current behaviour. If we later realise that it could be convenient to set a "default" builder for each type we can still figure out, which is the best place to set it.

So I would remove it here probably.